extensions/wikia/AdEngine/AdEngine2Hooks.class.php

```diff
@@ -255,4 +256,15 @@ public static function onVenusAssetsPackages( array &$jsHeadGroups, array &$jsBo
 		}
 		return true;
 	}
+
+	public static function onSkinTemplateOutputPageBeforeExec( Skin $skin, $template) {
+		$stagingHeader = $skin->getRequest()->getHeader('X-Staging');
+
+		if ( startsWith( $stagingHeader, static::ADS_STAGING_HEADER ) !== false ) {
+			$skin->getOutput()->setRobotPolicy( 'noindex,nofollow' );
+		}
```

Comment:
We do have [this](https://github.com/Wikia/app/blob/dev/extensions/wikia/Staging/Staging.setup.php)